services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -660,6 +664,93 @@ testScimCreateVsUserRef = do
       (call . delete $ brig . paths ["i", "users", toByteString' uid])
         !!! const 202 === statusCode
 
+testCreateUserTimeout :: TestSpar ()
+testCreateUserTimeout = do
+  env <- ask
+
+  email <- randomEmail
+  scimUser <- randomScimUser <&> \u -> u {Scim.User.externalId = Just $ fromEmail email}
+  (_owner, tid) <- call $ createUserWithTeam (env ^. teBrig) (env ^. teGalley)
+  tok <- registerScimToken tid Nothing
+
+  testFlow tid tok scimUser email
+  -- creating the same user again should behave the same
+  testFlow tid tok scimUser email
+  where
+    testFlow tid tok scimUser email = do
+      env <- ask
+      let brig = env ^. teBrig
+
+      scimStoredUser <- createUser tok scimUser
+      let handle = Handle . Scim.User.userName . Scim.value . Scim.thing $ scimStoredUser
+
+      inv <- call $ getInvitation brig email
+
+      -- Wait for user and invitation to expire
+      let setTeamInvitationTimeout = 5 -- Must match the value in brig.integration.yaml
+      threadDelay $ (setTeamInvitationTimeout + 1) * 1_000_000
+
+      -- User doesn't show in scim
+      do
+        listUsers tok (Just (filterBy "userName" $ fromHandle handle)) >>= \users ->
+          liftIO $ users `shouldBe` []
+        listUsers tok (Just (filterBy "externalId" $ fromEmail email)) >>= \users ->
+          liftIO $ users `shouldBe` []
+
+      -- User should fail to register invitation
+      do
+        call $ do
+          void $ do
+            Just inviteeCode <- getInvitationCode brig tid (inInvitation inv)
+            post
+              ( brig . path "/register"
+                  . contentJson
+                  . json (acceptWithName (Name "Bob") email inviteeCode)
+              )
+              <!! const 201 =/= statusCode
```

Comment:
```suggestion
              <!! const 400 === statusCode
```

...  or something.

weirdly, i just looked at the code, and couldn't see any error being thrown if the user tries to accept an invitation that doesn't exist any more.  i think there may be another bug in the code, where in this situation, a non-team users will be created and returned.

I may be missing something, good that we're writing this test!  :)  so, I say let's check for status code 999 and run the test, an we'll see what status code we actually get, and why.

Reply:
You get 400 with a body of
```
ponse {responseStatus = Status {statusCode = 400, statusMessage = "Bad Request"}, responseVersion = HTTP/1.1, responseHeaders = [("Transfer-Encoding","chunked"),("Date","Tue, 20 Oct 2020 16:24:30 GMT"),("Server","Warp/3.3.13"),("Content-Encoding","gzip"),("Content-Type","application/json")], responseBody = Just "{\"code\":400,\"message\":\"Invalid invitation code.\",\"label\":\"invalid-invitation-code\"}", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}
```
The error itself is defined to be 403
```
invalidCode = Wai.Error status403 "invalid-code" "Invalid verification code"
```
but his is as far as I got